bika/lims/browser/widgets/analysisspecificationwidget.py

```diff
@@ -35,152 +50,216 @@ def __init__(self, context, request, fieldvalue=[], allow_edit=True):
-                'sortable': False}),
-            ('rangecomment', {
-                'title': _('Range comment'),
-                'sortable': False,
-                'toggle': False}),
+            ("service", {
+                "title": _("Service"),
+                "index": "sortable_title",
+                "sortable": False}),
+            ("keyword", {
+                "title": _("Keyword"),
+                "sortable": False}),
+            ("methods", {
+                "title": _("Methods"),
+                "sortable": False}),
+            ("unit", {
+                "title": _("Unit"),
+                "sortable": False}),
+            ("service", {
+                "title": _("Service"),
+                "sortable": False}),
+            ("warn_min", {
+                "title": _("Min warn"),
+                "sortable": False}),
+            ("min", {
+                "title": _("Min"),
+                "sortable": False}),
+            ("max", {
+                "title": _("Max"),
+                "sortable": False}),
+            ("warn_max", {
+                "title": _("Max warn"),
+                "sortable": False}),
+            ("hidemin", {
+                "title": _("< Min"),
+                "sortable": False}),
+            ("hidemax", {
+                "title": _("> Max"),
+                "sortable": False}),
+            ("rangecomment", {
+                "title": _("Range comment"),
+                "sortable": False,
+                "toggle": False}),
         ))
 
         self.review_states = [
-            {'id':'default',
-             'title': _('All'),
-             'contentFilter':{},
-             'transitions': [],
-             'columns': self.columns.keys(),
-             },
+            {
+                "id": "default",
+                "title": _("All"),
+                "contentFilter": {"inactive_state": "active"},
+                "transitions": [],
+                "columns": self.columns.keys(),
+            },
         ]
 
+    def get_sorted_categories(self, items):
+        """Extracts the categories from the items
+        """
+
+        # XXX should be actually following the sortKey as well
+        categories = filter(
+            None, set(map(lambda item: item.get("category"), items)))
+
+        return sorted(categories)
+
+    def get_services(self):
+        """returns all services
+        """
+        catalog = api.get_tool("bika_setup_catalog")
+        query = self.contentFilter.copy()
+        # The contentFilter query get changed by the listing view to show only
+        # services in a category. This update ensures that the sorting is kept
+        # correct and that no inactive services are displayed.
+        query.update({
+            "inactive_state": "active",
+            "sort_on": self.sort_on,
+            "sort_order": self.sort_order,
+        })
+        logger.info("AnalysisSpecificationWidget::query=%r" % query)
+        return catalog(query)
```

Comment:
Remember the function `api.search(query, catalog)`, so there is no need of `api.get_tool` , etc. You could do just `return api.search(query, "bika_setup_catalog")`